### PR TITLE
[IMP] hr: add a model for departure reasons

### PR DESCRIPTION
**Pull request by kbapt**
The "Departure Reasons" where hardcoded and couldn't reflect all the
valid departure reasons (like "End Of Fixed-Term Contract").

This commit introduces a new model `hr.departure.reason`.

TaskID: 2447056

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr/data/hr_data.xml [hunk 1]
```diff
@@ -74,5 +74,25 @@
...
+
+        <record id="departure_retired" model="hr.departure.reason">
+            <field name="sequence">2</field>
+            <field name="name">Retired</field>
+        </record>
+
+        <record id="departure_freelance" model="hr.departure.reason">
+            <field name="sequence">3</field>
+            <field name="name">Became Freelance</field>
+        </record>
```
Comment (KangOl) [inline addons/hr/data/hr_data.xml:96]: I propose to add `Became Butcher`.

Joke aside, this departure reason seems too specific to IT. 
Plus, I don't see why we need to track the new occupation of old employees.

A more relevant departure reason would be `Death`.

Comment (kba-odoo) [reply addons/hr/data/hr_data.xml:96]: It's needed for the "Bilan Social", but it's indeed specific for Belgium... I'll move to the Belgian payroll instead.

Thanks :-)

**Comment by tivisse** [comment]
@robodoo r+

File: addons/hr/data/hr_data.xml (hunk 1 above)
Comment (tivisse) [reply addons/hr/data/hr_data.xml]: FYI, it's not to "track the new occupation of old employees", it's for employees who stopped their classic contract to start working as a freelance in the same company.

Comment (KangOl) [reply addons/hr/data/hr_data.xml]: It makes sense then.

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**